### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by pylang**
Changes
- Add project example
- Fix typos

Suggestions
- Line 107: what constitutes a "satisfied entry"?  Add a brief explanation.
- Line 152-5: consider a short explanation of each item somewhere
- Line 172: perhaps `local_dirpath` instead of `path` to remove ambiguity
- Line 202: similarly, `direct_url` instead of `direct`
- Line 279: `hashes` seems to be missing in the example

### CODE REVIEW

**Comment by pylang** [comment]
Poetry details how to [specify versions][0] a lock file.  Do the refs here clearly explain the version spec in this lock file?

  [0]: https://python-poetry.org/docs/dependency-specification/

**Comment by uranusjr** [comment]
Version range specification syntax is only relevant on locking time. The lock file records the result of the locking operation, and the specifiers are not an essential part of the lock file, since it is by design recording the result of those specifications. Poetry include specifiers in its lock file to (in a sense) make the file also double as a dependency cache, but IMO the feature is orthogotal to the lock file and does not need to be standardised as a part of it. The specification leaves enough room for tools like Poetry to add additional information to the file as they need.

File: README.md [hunk 1]
```diff
@@ -48,9 +49,9 @@ for a project.
 
 ### Specifying dependencies
 
-A top level key `dependencies` points to a mapping, which contains all
+Top-level key `dependencies` point to a mapping, which contain all
```
Comment (uranusjr) [inline README.md:52]: Maybe *A top-level key `dependencies` that points to a mapping  containing all [snip]*?

File: README.md [hunk 2]
```diff
@@ -71,8 +72,8 @@ bfrevity; they will be discussed in later sections):
 ```
 
 Each key in the mapping is a string that identifies a dependency. The key does
-not hold any special meanings itself, but used to refer the dependency it maps
-to, and should be unique in the lock file.
+not hold any special meaning itself, but the key is used to refer the dependency it maps
```
Comment (uranusjr) [inline README.md:75]: I believe *meanings* is an accepted form.

**Comment by pylang** [comment]
I appreciate the clarification on version specs; maybe add a brief note on why they are excluded (?).  As to "meaning" vs. "meanings", tomatoes tomatoes.  : )  

Brilliant draft on this.  I'm excited about the idea to standardize a lock file.  The doc mostly addresses details (which are indeed needed).  Some broad-spec questions that an outsider might have include:

- What is a lock file?
- Why do I need it?
- Does this replace requirements.txt?  If so, 
    - how do I (auto) convert requirements.txt to this lock file?
    - is it harder to use than requirements.txt?
- Does this improve real problems such as ...
      - slow dependency resolution?
      - namespace clobbering?
- etc.

The answers to such questions are not yet obvious from reading this.  Adding a short Motivation section might resolve this.

**Pull request merged**